backend/test_observer/controllers/test_results/test_results.py

```diff
@@ -0,0 +1,307 @@
+# Copyright (C) 2023 Canonical Ltd.
+#
+# This file is part of Test Observer Backend.
+#
+# Test Observer Backend is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# Test Observer Backend is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from datetime import datetime
+
+from fastapi import APIRouter, Depends, Query
+from sqlalchemy import and_, desc, distinct, select, func
+from sqlalchemy.orm import Session, joinedload
+
+from test_observer.data_access.models import (
+    Artefact,
+    ArtefactBuild,
+    Environment,
+    TestExecution,
+    TestResult,
+    TestCase,
+    Issue,
+    IssueTestResultAttachment,
+)
+from test_observer.data_access.models_enums import FamilyName
+from test_observer.data_access.setup import get_db
+
+from .models import TestResultSearchResponse
+
+router = APIRouter(tags=["test-results"])
+
+
+@router.get("", response_model=TestResultSearchResponse)
+def search_test_results(
+    family: str | None = Query(
+        None, description="Filter by artefact family (e.g., Charm, Snap)"
+    ),
+    environment: str | None = Query(
+        None,
+        description="Filter by environment name (e.g., Juju:3/stable ubuntu:20.04)",
+    ),
+    test_case: str | None = Query(
+        None, description="Filter by test case name (e.g., test_deploy)"
+    ),
+    template_id: str | None = Query(None, description="Filter by template ID"),
+    issues: str | None = Query(
+        None, description="Filter by Jira or GitHub issue IDs (comma-separated)"
+    ),
+    from_date: datetime | None = Query(
```

Comment:
I think either choose `from_date` and `until_date` or `from` and `until` for consistency with this naming.